### ISSUE TITLE
Avoid using EvalString

### ISSUE THREAD

**Issue by fingolfin** [gap-packages/Semigroups#463]
I just noticed that Semigroups (ab)uses `EvalString` quite a bit (I found 51 hits with grep). This is a function I'd recommend to never use, except for interactive debugging.

One place it is used in Semigroups is this:
```gap
for _IsXSemigroup in ["IsFpSemigroup",
                     "IsFpMonoid",
                     "IsNTPMatrixSemigroup",
                     "IsMaxPlusMatrixSemigroup",
                     "IsMinPlusMatrixSemigroup",
                     "IsTropicalMaxPlusMatrixSemigroup",
                     "IsTropicalMinPlusMatrixSemigroup",
                     "IsProjectiveMaxPlusMatrixSemigroup",
                     "IsIntegerMatrixSemigroup",
                     "IsReesMatrixSemigroup",
                     "IsReesZeroMatrixSemigroup"] do
  InstallMethod(TrivialSemigroupCons,
  Concatenation("for ", _IsXSemigroup, " and an integer"),
  [EvalString(_IsXSemigroup), IsInt],
  function(filter, deg)
    local n;
    n := Maximum(deg, 1);
    return AsSemigroup(filter,
                       TrivialSemigroupCons(IsTransformationSemigroup, deg));
  end);
od;
```

Here, a much better replacement is to use `ValueGlobal(_IsXSemigroup)`. It should even be faster.

Another case is `gap/tools/utils.gi`, which does `EvalString(Concatenation("IsBound(", name, ")")) `. I suspect that there, you can get aways with using `IsBoundGlobal(name)` instead. 

This leaves the tests, where your use of `EvalString` seems OK.

**Comment by james-d-mitchell** [comment]
Thanks @fingolfin will fix this as soon as I am able.

**Comment by james-d-mitchell** [comment]
This is resolved via PR #464. Thanks for pointing this out @fingolfin
